### ISSUE TITLE
[chip] accessibility issues relating to delete button on input chips

### ISSUE THREAD

**Issue by DRiFTy17** [tyler-technologies-oss/forge#184]
**Describe the bug:**
The delete button does not announce properly with NVDA. It reads "blank" because the `aria-hidden="true"` attribute tells the screen reader to ignore it. This attribute is added automatically via the `<forge-icon>` element.

Additionally the screen reader announces "Delete button button" in this case because the delete icon button is nested within the outer `<button>` container which is incorrect. We should also remove this nesting to avoid this.

**Expected behavior:**
it should announce "Delete" as specified on the `aria-label`.

**Please complete the following information:**
 - Forge version: ^2.0.0
 - I have searched existing issues before creating this report? Y
 - Browser: Chrome
 - Platform: Desktop
 - OS: Windows
 - Screen reader: NVDA

**Additional context:**
Reported internally here:
https://tylerjira.tylertech.com/browse/TCPSD-13851
